keps/sig-apps/20190523-support-completions-index-for-job.md

```diff
@@ -0,0 +1,78 @@
+---
+title: support-completions-index-for-job
+authors:
+  - "@fudali"
+owning-sig: sig-apps
+participating-sigs:
+  - sig-job
+reviewers:
+  - "@kow3ns"
+  - "@janetkuo"
+approvers:
+  - "@kow3ns"
+  - "@janetkuo"
+editor: TBD
+creation-date: 2019-05-23
+last-updated: 2019-05-27
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# support completions index for JOB
+
+## Table of Contents
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+- [Proposal](#proposal)
+  - [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+
+
+## Summary
+
+[Job](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/)
+is a Kubernetes CRD that help us run task, Job allows a user to specify the task or parallel task through either `podTemplate` and `completions`; but i use parallel task i must ensure task unrepeat in myself application, this is diffcult; if job support sahrds, i can use shards index to implement task unrepeat is so easy;
```

Comment:
Can you give an example on what's kind of workload is running in the Pod, e.g. MPI, Spark? If it's pure customized application, why not crd + operator? If

Reply:
btw, we implemented similar feature by CRD+Operator at https://github.com/volcano-sh/volcano/blob/master/docs/design/job-api.md#plugins-for-job which maybe helpful for your case.

Reply:
such as we have million user, each user has different user results, i need push one article to user where has some result。  

we expected run these task in one platform，that platform must easy to use and in common use，so we choice k8s；and we don`t want to use more platform(because we need lower maintenance cost)

solution is more， but i expected i need be for maintenance do something is less and i see this feature is not implement in document. i think this feature can help more people easy to use k8s.(we company use [argo](https://github.com/argoproj/argo) to implement we case, but [argo](https://github.com/argoproj/argo) more feature not to use)

Reply:
is there others have similar case with yours?

Reply:
we company has this case and i think this is a common question. because this implement can make job application easy to scale.

but my issue not more people comment support this implement.

@k82cn

Reply:
@fudali113 , IMO, that's a specific case, and prefer to handle that by CRD :)